queue.go

```diff
@@ -165,6 +166,7 @@ func (q *Queue) Put(val interface{}, timeout float64) error {
 	q.clearPending()
 	isfull := q.isfull()
 	if timeout < 0.0 && isfull {
+		q.mutex.Unlock()
```

Comment:
Yes, that a bug, thank you to fix it, but your test case does not cover the code that you added.